android/build.gradle

```diff
@@ -1,24 +1,52 @@
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
 
 buildscript {
-    repositories {
-        jcenter()
+    // The Android Gradle plugin is only required when opening the android folder stand-alone.
+    // This avoids unnecessary downloads and potential conflicts when the library is included as a
+    // module dependency in an application project.
+    if (project == rootProject) {
+        repositories {
+            google()
+            jcenter()
+        }
+        dependencies {
+            classpath 'com.android.tools.build:gradle:3.5.2'
+        }
     }
-    dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+}
+
+apply plugin: 'com.android.library'
 
-        // NOTE: Do not place your application dependencies here; they belong
-        // in the individual module build.gradle files
+android {
+    compileSdkVersion safeExtGet('compileSdkVersion', 28)
+    buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
+    defaultConfig {
+        minSdkVersion safeExtGet('minSdkVersion', 16)
+        targetSdkVersion safeExtGet('targetSdkVersion', 28)
```

Comment:
I don't think we want to allow overriding of `minSdkVersion` and `targetSdkVersion` from the root project. These two variables declare the Android API that this library has been coded against. Decreases of `minSdkVersion` should only be done after the library has been updated to not use any APIs not available in the older version. Likewise, increases of `targetSdkVersion` should only be done after the library has been updated to account for any runtime behavior changes introduced in the newer version. Otherwise the app can experience crashes at runtime while using this library.

The Android Gradle plugin already includes behavior to merge the app's version of these two variables with the values of these variables from all of the app's library dependencies, with proper safety checks. And if an app wants to override one of the built-in safety checks it can do so via manifest markers described at https://developer.android.com/studio/build/manifest-merge#override_uses-sdk_for_imported_libraries.

Reply:
I'm not an android developer, so I have no clue what I am talking about here™️  

On the React Native side of things, I _think_ this is done just to try and keep up with the minimum versions that RN supports.  I am no expert on the non-js parts of RN, but my guess would be that it is fine to remove it. I will test it and update.